### PR TITLE
docs(core): sync nx package versions

### PR DESCRIPTION
**Pull request by isaacplmann**
Adds messaging to keep Nx versions in sync

### CODE REVIEW

File: docs/shared/core-features/automate-updating-dependencies.md [hunk 1]
```diff
@@ -66,6 +66,28 @@ After you run all the migrations, you can remove `migrations.json` and commit an
 
 Note: You may want to keep the `migrations.json` until every branch that was created before the migration has been merged. Leaving the `migrations.json` in place allows devs to run `nx migrate --run-migrations` to apply the same migration process to their newly merged code as well.
 
+## Keep Nx Packages on the Same Version
```
Comment (barbados-clemens) [inline docs/shared/core-features/automate-updating-dependencies.md:69]: should we add a link to this page inside the call outs added to each plugin page?

Comment (barbados-clemens) [reply docs/shared/core-features/automate-updating-dependencies.md:69]: or maybe at least some info about 'run nx report to see what versions are out of sync easily'.

**Review by barbados-clemens — APPROVED** [review]

File: docs/shared/core-features/automate-updating-dependencies.md (hunk 1 above)
Comment (isaacplmann) [reply docs/shared/core-features/automate-updating-dependencies.md:69]: Good idea.

**Review by AgentEnder — APPROVED** [review]

**Review by ndcunningham — APPROVED** [review]

File: docs/shared/core-features/automate-updating-dependencies.md [hunk 2]
```diff
@@ -66,6 +66,28 @@ After you run all the migrations, you can remove `migrations.json` and commit an
...
+
+When you run `nx migrate`, the `nx` package and all the `@nx/` packages get updated to the same version. It is important to keep these versions in sync, or you can encounter some difficult to debug errors. As long as you run `nx migrate` instead of manually changing the version numbers, you shouldn't have to worry about it.
+
+{% callout type="note" title="Use the latest nx-cloud version" %}
+The `nx-cloud` package does not need to be in sync with the other Nx packages. For best results, stay on the latest version of `nx-cloud`. The latest `nx-cloud` version supports the most recent 2 major versions of `nx`, although earlier versions of `nx` may also be compatible.
+{% /callout %}
+
+## Migrate Community Plugins Individually
+
+The `nx migrate` command migrates all the official plugins, but not the community plugins you have installed. Those plugins should be migrated individually with:
```
Comment (FrozenPandaz) [inline docs/shared/core-features/automate-updating-dependencies.md:79]: I think this should be rephrased.

To me, this sounds like `nx migrate` *should* migrate community plugins but does not. I think it should be more positive in its wording to say `nx migrate` also works for community plugins and should be used to update those as well.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: docs/generated/packages/angular/documents/overview.md [hunk 3]
```diff
@@ -27,6 +27,10 @@ yarn add -D @nx/angular
 npm install -D @nx/angular
 ```
 
+{% callout type="note" title="Keep Nx Package Versions In Sync" %}
+Make sure to install the `@nx/angular` version that matches the version of `nx` in your repository. If the version numbers get out of sync, you can encounter some difficult to debug errors. Run `nx report` to see a list of installed plugins. Use [`nx migrate`](/core-features/automate-updating-dependencies) to update plugin versions.
+{% /callout %}
```
Comment (Coly010) [inline docs/generated/packages/angular/documents/overview.md:32]: Do we need to call out here that if you already have a a higher version of one of the packages, you may need to run `nx migrate --from --to` or are we expecting the user to get that information from the migrate docs?

**Review by Coly010 — APPROVED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Review by juristr — CHANGES_REQUESTED** [review]

**Review by meeroslav — APPROVED** [review]

**Review by juristr — APPROVED** [review]

**Pull request merged**